mule-apikit-tools/mule-apikit-scaffolder/src/test/java/org/mule/tools/apikit/VersionUtilsTest.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit;
+
+import org.junit.Test;
+import static org.mule.tools.apikit.misc.VersionUtils.getMinVersion;
+import static org.junit.Assert.assertEquals;
+
+public class VersionUtilsTest {
+
+  private static final String minVersion = "4.1.4";
+
+  @Test
+  public void versionUtilsTests() {
+
+    String version;
+
+    version = getMinVersion("4.0.0", minVersion);
+    assertEquals(minVersion, version);
+
+    version = getMinVersion(null, minVersion);
+    assertEquals(minVersion, version);
+
+    version = getMinVersion("Not Valid Version", minVersion);
+    assertEquals(minVersion, version);
+
+    version = getMinVersion("4.2", minVersion);
+    assertEquals(minVersion, version);
```

Comment:
In this case, shouldn't the returned version be "4.2"?

Reply:
Maybe it should, but I think that mule versions must have exactly 3 parts, so I've simplified that way. There is also an Apache Library for versioning , but I think it would be an overkill.

Reply:
Ok, I see your point. No need to use Apache Library! I agree with you so.
Some extra comment, the method name is a little misleading, considering that the name is `getMinVersion` when actually it returns the maximum one

Reply:
Agree, changed `getMaxVersion`.